src/saveload/saveload.cpp

```diff
@@ -1217,6 +1217,67 @@ static void *IntToReference(size_t index, SLRefType rt)
 	}
 }
 
+/**
+ * Template class to help with list-like types.
+ */
+template <template<typename, typename> typename Tstorage, typename Tvar, typename Tallocator = std::allocator<Tvar>>
+class SlStorageHelper {
+	typedef Tstorage<Tvar, Tallocator> SlStorageT;
+public:
+	/**
+	 * Internal templated helper to return the size in bytes of a list-like type.
+	 * @param storage The storage to find the size of
+	 * @param conv VarType type of variable that is used for calculating the size
+	 */
+	static size_t SlCalcLen(const void *storage, VarType conv)
+	{
+		const SlStorageT *l = (const SlStorageT *)storage;
+
+		int type_size = 4;
```

Comment:
This feels kind of a "random" name. It's not really the type size, but it is already super unclear in the original code due to the bad comment.
I know it's the size in (8 bit) bytes of the uint32 size field that is being stored in the save game. But naming that clearly and concisely is a bit weird; the obvious one would be `size_size`, i.e. the size of the size, but that makes no real sense. Similarly `length_size`, but it's the `sizeof` of the result of `size()` that is written.

Maybe `return sizeof(uint32) + l->size() * item_size;`? That would at least tell you there a `uint32` + `size()` times something of `item_size`, and implicitly I'd reckon that would imply "a size field + all the elements".

Reply:
Again, moved code. This rabbit-hole is already rather deep, so I really want to scope it at some point. I don't want to second-guess every decision ever made in our code-past :D

Reply:
For now I settled on a comment. Soon (tm) the line is going to be removed anyway, as it is stupid that we spend 4 bytes on a length-indicator and not use a gamma, where 80% of the rest of the code just uses a gamme for length-indicators .. ugh .. SaveLoad code :P So a comment will have to do for now :)